src/main/java/com/ldtteam/blockout/views/TabSet.java

```diff
@@ -0,0 +1,271 @@
+package com.ldtteam.blockout.views;
+
+import com.ldtteam.blockout.Alignment;
+import com.ldtteam.blockout.Pane;
+import com.ldtteam.blockout.PaneParams;
+import com.ldtteam.blockout.controls.Image;
+import com.ldtteam.blockout.controls.Text;
+import com.mojang.blaze3d.matrix.MatrixStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.Tuple;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Random;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Similar to a switch view, the tab set renders the selected view,
+ * allowing for tabs to be placed for user's selection
+ */
+public class TabSet extends View
+{
+    private static final Random RANDOM = new Random();
+
+    protected boolean vertical;
+    protected int spacing;
+    protected int overlap;
+    protected int offset;
+    protected int tabWidth;
```

Comment:
this seems related to buttonTab, not just tab

Reply:
They sure are related to tabs. Can prefix these to make it clearer, but I'm not going to write XML attributes in every tab that are all the same if I can just write them in the tabview instead.